jest/preprocessor.js

```diff
@@ -56,6 +56,7 @@ module.exports = {
         inlineRequires: true,
         minify: false,
         platform: '',
+        babelrc: false,
```

Comment:
uhm maybe this fixes https://github.com/facebook/react-native/issues/19859? Not sure

Reply:
`babelrc: false` was added so that the `babel.config.js` file is ignored by the jest preprocessors transform logic in this file - this babel config file is only needed for `local-cli` whilst running inside CI  & local tests. 

The jest preprocessor requires it's own babel plugins here: https://github.com/facebook/react-native/blob/3af30dfcc73a8fb879c55e573b4072b2f591b33c/jest/preprocessor.js#L65 - so we don't want the `babel.config.js` file to interfere with this otherwise it loads RN specific Babel plugins and causes all sorts of havoc.

Reply:
Can you add more context here? Maybe include a comment at the top of the Babel.config.js?

What needs it and what needs to not use it?

From your comment I couldn’t tell which thing you are saying jest needs.

Reply:
@TheSavior, `babelrc: false` was added to prevent the Jest tests in the React Native project from using the `babel.config.js` file which was added at the root of the project - this new config file is only for `local-cli` usage whilst testing locally and in CI - without this config file Metro will bundle RN app bundles for the test apps without using the `metro-react-native-babel-preset` preset - which was causing the CI failures..

Without this option (`babelrc: false`) the custom Jest preprocessor for this project ([here](https://github.com/facebook/react-native/blob/master/jest/preprocessor.js)) will read the additional config in `babel.config.js` and subsequently merge that config into the configuration defined in the `transformer.transform` options - https://github.com/facebook/react-native/blob/master/jest/preprocessor.js#L50.

We don't want this to happen as all the babel plugins defined in `metro-react-native-babel-preset` are added to the custom Jest preprocessor - on top of the one already defined in the `transformer` configuration.

Reply:
@TheSavior have also just added a descriptive comment header on the config file: https://github.com/facebook/react-native/pull/20854/commits/6ad1be5d12fe25db84c6f20c12827d19963bf8f0